pkg/landscaper/jsonschema/reference.go

```diff
@@ -132,18 +135,26 @@ func (rr *referenceResolver) resolveList(data []interface{}, currentPath *field.
 }
 
 // resolveReference resolves a reference
-func (rr *referenceResolver) resolveReference(s string, currentPath *field.Path) (interface{}, error) {
+func (rr *ReferenceResolver) resolveReference(s string, currentPath *field.Path, alreadyResolved stringSet) (interface{}, error) {
 	uri, err := url.Parse(s)
 	if err != nil {
 		return nil, fmt.Errorf("invalid url: %w", err)
 	}
+	if rr == nil || rr.ReferenceContext == nil {
```

Comment:
I don't understand why this is checked here.
`rr` is the this pointer and is used in a non-exported "private" member function of `ReferenceResolver`.
I don't see why it can be nil at this point.

For the `rr.ReferenceContext`, this field is getting initialized in `NewReferenceResolver` if it is passed as nil by the user.
It can only be nil when someone explicitely sets the pointer to nil.
If it needs to be checked for nil, doesn't it make more sense to do it in function `Resolve`?

Reply:
I did this to avoid some problems with the tests: 
Before, rr.ReferenceContext being nil was no problem, but after I added the cycle check, it didn't work anymore. Therefore I added this check and I forgot to remove it, after I added the defaulting in the constructor. I guess we don't need this check anymore.